tests/test_etl.py

```diff
@@ -64,42 +70,50 @@ def get(self, url, params):
 
         dcplib.etl.http = MockDSSClient()
         with tempfile.TemporaryDirectory() as td:
-            e = dcplib.etl.DSSExtractor(staging_directory=td,
-                                        content_type_patterns=["application/json"],
-                                        filename_patterns=["*.json"],
-                                        dss_client=MockDSSClient())
-            e.extract(transformer=tf,
-                      loader=ld,
-                      finalizer=fn,
-                      query={"test": True},
-                      max_workers=2,
-                      max_dispatchers=1)
+            e = dcplib.etl.DSSExtractor(
+                staging_directory=td,
+                content_type_patterns=["application/json"],
+                filename_patterns=["*.json"],
+                dss_client=MockDSSClient(),
+                transformer=tf,
+                loader=ld,
+                finalizer=fn
+            )
+            e.etl_bundles(query={"test": True},
+                          max_workers=2,
+                          max_dispatchers=1)
             self.assertEqual(calls["tf"], 4)
-            self.assertEqual(calls["ld"], 4)
+            self.assertEqual(calls["ld"], 1)
```

Comment:
You have changed the loader to be called once at the end of the process with all the bundles as the argument. Is there a reason why you did this? The change negates the interleaving advantage that processing one bundle at a time confers - you won't start loading until after all the extractions are complete, so the database load and latency is not amortized.

I think it's preferable to go back to loading one bundle at a time.

Reply:
discussed in person